<launch rosmon-name="rosmon_uut">
	<!-- Test $(find ...) -->
	<param name="path_to_rosmon" value="$(find rosmon)" />
	<param name="path_to_launch_file" value="$(find rosmon)/test/basic.launch" />
	<param name="path_to_rosmon_executable" value="$(find rosmon)/rosmon" />

	<!-- Test $(dirname) -->
	<param name="dirname" value="$(dirname)" />

	<!-- Test $(eval) -->
	<param name="eval_simple" value="$(eval True)" />

	<arg name="arg1" value="true" />
	<arg name="arg2" value="false" />
	<param name="eval_argexpr" value="$(eval arg1 and not arg2)" />

	<arg name="radius" value="0.5" />
	<param name="eval_radius_pi" value="$(eval radius * pi)" />

	<!-- Test launch file inclusion -->
	<arg name="test_argument" default="123" />
	<include file="$(dirname)/basic_sub.launch" pass_all_args="true" />

	<!-- Test params over multiple lines -->
	<param name="multiple_lines1" value="first_line
	second_line" />
	<param name="multiple_lines2" command="echo -n first_line
	second_line" />

	<!-- Test YAML params -->
<rosparam subst_value="true">
yaml:
  radius: $(arg radius)
</rosparam>

	<!-- Test loading of empty YAML files/strings
	  (see https://github.com/xqms/rosmon/pull/12 for discussion) -->
	<rosparam command="load" file="$(dirname)/empty.yaml" />
	<rosparam>
	</rosparam>

	<node name="test1" pkg="rosmon" type="test_node.py">
		<remap from="~input" to="/test_input" />
		<remap from="~output" to="/test_output" />

		<param name="private_param1" value="hello_world" />
		<param name="~private_param2" value="hello_world" />
	</node>

	<!-- Test global remapping -->
	<remap from="original_test_input" to="remapped_test_input" />
	<remap from="original_test_output" to="remapped_test_output" />
	<node name="test2" pkg="rosmon" type="test_node.py">
			<remap from="~input" to="/original_test_input" />
			<remap from="~output" to="/original_test_output" />
	</node>
</launch>
